src/main/java/kostka/moviecatalog/service/rabbitmq/RabbitMqReceiver.java

```diff
@@ -96,4 +103,26 @@ public void receiveUpdateRequestRecalculateQueue() {
             LOGGER.error("Completable future error?", e);
         }
     }
+
+    @RabbitListener(
+            bindings = @QueueBinding(
+                    exchange = @Exchange(TOPIC_EXCHANGE),
+                    key = RATING_KEY,
+                    value = @Queue(RATING_QUEUE)
+            )
+    )
+    public void receiveRecalculateAverageMovieRating() {
+        LOGGER.info("Received message from RabbitMQ to get Ratings for all movies.");
+        statisticService.incrementSyncedRabbitMqCounter();
+        List<Movie> allMovies = dbMovieService.getAllMoviesFromDB();
+        allMovies.parallelStream().forEach(
```

Comment:
Jen poznámka: parallelStream není zas tak optimalizovaný, jak by mohl. Někdy dokonce může performance zhoršit (overhead, synchronizace, ...). Někde jsem četl doporučení, že by se měl použít až když máš nějaký benchmark test a víš, že se opravdu oplatí využít.

Je vhodné použít až:
- máš opravdu velké množství dat
- víš, že máš problém s rychlostí s použitím obyčejného streamu
- máš benchmark

Reply:
Hm, např. jeden článek https://dzone.com/articles/think-twice-using-java-8

Reply:
dam foreach, diky za clanek :)